src/main/java/hudson/plugins/warnings/parser/DrMemoryParser.java

```diff
@@ -0,0 +1,136 @@
+package hudson.plugins.warnings.parser;
+
+import hudson.Extension;
+import hudson.plugins.analysis.util.model.Priority;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A parser for the Dr. Memory Errors.
+ *
+ * @author Wade Penson
+ */
+@Extension
+public class DrMemoryParser extends RegexpDocumentParser {
+    private static final long serialVersionUID = 7195239138601238590L;
+
+    /** The index of the regexp group capturing the header of the error. */
+    private static final int HEADER_GROUP = 1;
+
+    /** The index of the regexp group capturing the body of the error or warning. */
+    private static final int BODY_GROUP = 2;
+
+    private static final String DR_MEMORY_WARNING_PATTERN =
+        "Error #\\d+:[ ]*(.+)\\s*([\\s\\S]*?\\n)(?:\\n|[^#])";
+
+    /** Pattern to extract file paths from body. */
+    private static final Pattern FILE_PATHS_PATTERN =
+        Pattern.compile("#\\s*\\d+\\s*.*?\\[(.*\\/.*):(\\d+)\\]");
+
+    /** Pattern to extract jenkins path from file path. */
+    private static final Pattern JENKINS_PATH_PATTERN =
+        Pattern.compile(".*?\\/jobs\\/.*?\\/workspace\\/");
+
+    /**
+     * Creates a new instance of {@link DrMemoryParser}.
+     */
+    public DrMemoryParser() {
+        super(Messages._Warnings_DrMemory_ParserName(),
+                Messages._Warnings_DrMemory_LinkName(),
+                Messages._Warnings_DrMemory_TrendName(),
+                DR_MEMORY_WARNING_PATTERN, true);
```

Comment:
This only applies when RegexpLineParser is extended by the looks of it but the RegexpDocumentParser was extended instead since warnings span multiple lines. I used the Doxygen and Eclipse parser as reference.